docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,254 @@
+  - **Mitigation:** The processes are automated within the controller, requiring minimal operator intervention. Clear documentation and tooling support will help manage complexity.
+
+- **Failed Snapshot Before Hibernation:**
+  - **Risk:** Failure to take a full snapshot before hibernation could delay the hibernation process and potentially compromise data integrity.
+    - **Mitigation:** Implement robust error handling and retry mechanisms to ensure snapshots are taken successfully before hibernation. Notify operators of any failures by updating the `etcd.status.lastErrors` and `etcd.status.lastOperation` fields. Additionally, operators can leverage the metrics provided by `ExtendEtcdSnapshotImmutabilityTask`, which follows the [operator task framework](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/05-etcd-operator-tasks.md#metrics), to trigger alerts. This ensures timely intervention and resolution of issues, maintaining the integrity and availability of the etcd cluster state.
+
+## Alternatives
+
+### Object-Level Immutability Policies vs. Bucket-Level Immutability Policies
+
+An alternative to implementing immutability via bucket-level immutability policies is to use object-level immutability policies. Object-level immutability allows for more granular control over the immutability periods of individual objects within a bucket, whereas bucket-level immutability applies a uniform immutability period to all objects in the bucket.
+
+#### Feasibility Study: Immutable Backups on Cloud Providers
+
+Major cloud storage providers such as Google Cloud Storage (GCS), Amazon S3, and Azure Blob Storage (ABS) support both bucket-level and object-level immutability mechanisms to enforce data immutability.
+
+1. **Bucket-Level Immutability Policies:**
+   - **Applies Uniformly:** Applies a uniform immutability period to all objects within a bucket.
+   - **Immutable Objects:** Once set, objects cannot be modified or deleted until the immutability period expires.
+   - **Simplified Management:** Simplifies management by applying the same policy to all objects.
+
+2. **Object-Level Immutability Policies:**
+   - **Granular Control:** Allows setting immutability periods on a per-object basis.
+   - **Flexible Immutability Durations:** Offers granular control, enabling different immutability durations for individual objects.
+   - **Varying Requirements:** Can accommodate varying immutability requirements for different types of backups.
+
+#### Considerations for Object-Level Immutability
+
+Using object-level immutability provides flexibility in scenarios where certain backups require different immutability periods. However, current limitations and complexities make it less practical for immediate implementation.
+
+- Enabling object-level immutability requires bucket-level immutability to be set first (applicable in S3 and ABS). In GCS, the capability to enable object-level immutability on an existing bucket is not available.
```

Comment:
Can you please also mention the plan/date by which GCS is going to enable object level immutability. If there is an issue open please provide a link here.